nautobot/core/views/generic.py

```diff
@@ -232,6 +233,9 @@ def get(self, request):
                 )
                 self.queryset = self.queryset.none()
 
+            if filterset.is_valid() and filterset.data:
+                hide_hierarchy_ui = True
```

Comment:
Just wondering - what part of this only applies if a filter query parameter is present? Does `filterset.data` not get set if there are no filters?

Reply:
An empty filterset query-param is also valid filterset and we dont want to hide hierarchy indentation if no filter query-param is provided.

Reply:
No, the goal is understood - I was asking how this logic actually enables the desired behavior.

Reply:
I pass the `hide_hierarchy_ui` into the instantiation of the table

```python
table = self.table(self.queryset, user=request.user, order_by=order_by, hide_hierarchy_ui=hide_hierarchy_ui)
```

Then use it here:
```
TREE_LINK = """
 {% load helpers %}
 {% tree_hierarchy_ui_representation record.tree_depth|as_range table.hide_hierarchy_ui %}
 <a href="{{ record.get_absolute_url }}">{{ record.name }}</a>
 """
```

Does this answer the question?

Reply:
`filterset.data` is where the filter params are passed in here:

https://github.com/nautobot/nautobot/blob/8d2cf2cd698e31bd69427b4dc88bc02ba5a8cad7/nautobot/core/views/generic.py#L217-L219

`data` is the first arg to `BaseFilterSet.__init__`